include/mbedtls/pk.h

```diff
@@ -361,24 +361,35 @@ int mbedtls_pk_setup(mbedtls_pk_context *ctx, const mbedtls_pk_info_t *info);
 /**
  * \brief           Initialize a PK context to wrap a PSA key.
  *
- * \note            This function replaces mbedtls_pk_setup() for contexts
- *                  that wrap a (possibly opaque) PSA key instead of
- *                  storing and manipulating the key material directly.
+ *                  This function helps creating a PK context which wraps a
+ *                  PSA key. The PSA wrapped key must:
+ *                  * remain valid as long as the wrapping PK context is in use,
+ *                    that is at least between the point this function is
+ *                    called and the point mbedtls_pk_free() is called on this
+ *                    context;
+ *                  * be a key pair;
+ *                  * be an EC or RSA type (DH is not suported in PK module).
+ *
+ *                  Under the hood PSA functions are used to perform the required
+ *                  operations and, based on the key type, used algorithms will be:
+ *                  * EC:
+ *                      * verify, verify_ext: #PSA_ALG_ECDSA_ANY;
+ *                      * sign, sign_ext: try #PSA_ALG_DETERMINISTIC_ECDSA()
+ *                        first and, in case it fails, try with #PSA_ALG_ECDSA().
```

Comment:
“try … in case it fails” is an implementation detail. What we promise is `#PSA_ALG_DETERMINISTIC_ECDSA` if supported and `#PSA_ALG_ECDSA` otherwise.

But should we actually promise this? For non-opaque keys, we don't currently specify which ECDSA variant is used. (The preference for deterministic when available is documented for `mbedtls_ecdsa_write_signature`, but we don't document that `mbedtls_pk_sign` calls that function.)